compiler.rst

```diff
@@ -159,49 +159,81 @@ are very rare.  However, if you've allocated a PyObject, you must tell
 the arena about it by calling ``PyArena_AddPyObject()``.
 
 
-Parse Tree to AST
------------------
-
-The AST is generated from the parse tree (see :file:`Python/ast.c`) using the
-function ``PyAST_FromNode()``.
+Tokens to AST
+-------------
 
-The function begins a tree walk of the parse tree, creating various AST
-nodes as it goes along.  It does this by allocating all new nodes it
-needs, calling the proper AST node creation functions for any required
-supporting functions, and connecting them as needed.
+The AST is generated from tokens using the function
+``_PyPegen_run_parser()`` (:file:`Parser/pegen.c`), which is a thin wrapper over
+the function ``_PyPegen_parse()`` (:file:`Parser/parser.c`).
```

Comment:
`_PyPegen_run_parser` and `_PyPegen_parse` are technically implementation details. I prefer to reference the actual APis used by the interepter: `_PyParser_ASTFromFile` and `_PyParser_ASTFromString` (and other variants).